### PR TITLE
Feature #53 feat: Added new attribute to file type and introduced new type for video links

### PR DESCRIPTION
**Pull request by praneettekdi**

### CODE REVIEW

File: administrator/models/fields/video.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	 * @return  array
+	 *
+	 * @since 3.7
+	 */
+	protected function getLayoutData()
+	{
+		$data = parent::getLayoutData();
+
+		// Initialize some field attributes.
+		$maxLength    = !empty($this->maxLength) ? ' maxlength="' . $this->maxLength . '"' : '';
```
Comment (ankush-maherwal) [inline administrator/models/fields/video.php]: do we need this?

As Youtube and Vimeo videos length are generated by them, we can't decide maxlength for the links

File: administrator/models/fields/video.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+		// Note that the input type "url" is suitable only for external URLs, so if internal URLs are allowed
+		// we have to use the input type "text" instead.
+		$inputType    = $this->element['relative'] ? 'type="text"' : 'type="url"';
```
Comment (ankush-maherwal) [inline administrator/models/fields/video.php]: @praneettekdi 
do we need this?

File: administrator/models/forms/types/forms/video.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<form>
+	<fields name="params" label="COM_TJFIELDS_FIELD_BASIC_LABEL">
+		<fieldset name="params">
+			<field name="maxlength" type="text" label="COM_TJFIELDS_FORM_LBL_FIELD_MAXLENGTH" description="COM_TJFIELDS_FORM_DESC_FIELD_SIZE"/>
+			<field name="rendervideolink" type="radio" default="0" label="COM_TJFIELDS_FORM_LBL_FIELD_RENDER_VIDEO_LINK" description="COM_TJFIELDS_FORM_DESC_FIELD_RENDER_VIDEO_LINK" class="btn-group">
```
Comment (ankush-maherwal) [inline administrator/models/forms/types/forms/video.xml]: @praneettekdi 
setting `rendervideolink` to yes will load the video in iframe right?

But the name suggests that video link will be rendered.

Can we change this to `rendervideo` or something

File: administrator/models/fields/video.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+			$videoUrl = 'https://www.youtube.com/embed/' . $isValidYoutubeUrl;
+		}
+		elseif ($isValidVimeoUrl)
+		{
+			$videoUrl = 'https://player.vimeo.com/video/' . $isValidVimeoUrl;
+		}
+
+		if (!empty($videoUrl) && $layoutData['field']->element->attributes()->rendervideolink)
+		{
+			$html .= '<iframe width="320" height="240" src="' . $videoUrl . '"></iframe>';
```
Comment (ankush-maherwal) [inline administrator/models/fields/video.php]: @praneettekdi 
can we pass height and width as field params?
user should be able to change the height and width as per the requirement

File: site/helpers/tjfields.php [hunk 5]
```diff
@@ -1732,4 +1732,45 @@ public static function getLanguageConstantForJs()
...
+	public function isValidYoutubeUrl($url)
+	{
+		$rx = '~
+		^(?:https?://)?                          # Optional protocol
+		(?:www[.])?                              # Optional sub-domain
+		(?:youtube[.]com/watch[?]v=|youtu[.]be/) # Mandatory domain name (w/ query string in .com)
+		([^&]{11})                               # Video id of 11 characters as capture group 1
+		~x';
+
+		$has_match = preg_match($rx, $url, $matches);
```
Comment (ankush-maherwal) [inline site/helpers/tjfields.php]: @praneettekdi 
Please check issues reported by scrutinizer

File: administrator/models/fields/video.php [hunk 6]
```diff
@@ -0,0 +1,101 @@
+<?php
```
Comment (coolbung) [inline administrator/models/fields/video.php]: Is there a doc describing how the video field should work ? This field seems to support just youtube, vimeo and "renderurl" ? 

The video field design needs a discussion. There are a few approaches that we can take
- One field per type
- Single field using videojs (in this case the video type will need to be auto detected from the URL?)

The field attributes also need to be carefully designed.

File: site/helpers/tjfields.php (hunk 5 above)
Comment (coolbung) [reply site/helpers/tjfields.php]: This method is needed by a specific field type.. so it cannot be in the core. 

Anyway.. check my comment about the video field in general. The implementation seems too specific.

**Comment by ankush-maherwal** [comment]
MR is obsolete and can be closed

**Pull request closed without merging**